### ISSUE TITLE
no-extra-parens conflicts with arrow function returning object

### ISSUE THREAD

**Issue by RReverser** [babel/babel-eslint#129]
In order to return object from arrow function, one usually uses following syntax:
```javascript
var func = param => ({
	a: 1,
	b: 2
});
```

Since curly braces after arrow mean beginning of block otherwise.

However, ESLint + babel-eslint + `no-extra-parens` rule treat this as unnecessary parentheses while it's definitely required one.

**Comment by hzoo** [comment]
@RReverser can you reproduce with the regular ESLint parser? Looks like it errors for that as well

**Comment by RReverser** [comment]
@hzoo Probably, didn't try. Should I report to their repo instead?

**Comment by hzoo** [comment]
@RReverser Yeah if you can try it and comment out `"parser": "babel-eslint",` with `"es6": true` and it still error's (it is for me) then I think it should go in the ESLint repo.

**Comment by RReverser** [comment]
Thanks, moved to https://github.com/eslint/eslint/issues/2749.
